scripts/update_prescriptions.py

```diff
@@ -169,9 +230,150 @@ def update_examples(api_base_url):
+                    prescription_id_updated = extension["valueIdentifier"]["value"]
+    assert prescription_id == prescription_id_updated
+
+
+def test_update_prescription__updates_short_prescription_id(secondary_care_repeat_dispensing_process_request):
+    short_prescription_id = "newValue"
+    update_prescription(secondary_care_repeat_dispensing_process_request, None, short_prescription_id, None, None)
+    for entry in secondary_care_repeat_dispensing_process_request['entry']:
+        resource = entry["resource"]
+        if resource["resourceType"] == "MedicationRequest":
+            short_prescription_id_updated = resource["groupIdentifier"]["value"]
+    assert short_prescription_id == short_prescription_id_updated
+
+
+def test_update_prescription__updates_signature_time(secondary_care_repeat_dispensing_process_request):
+    signature_time = "newValue"
+    update_prescription(secondary_care_repeat_dispensing_process_request, None, None, None, signature_time)
+    for entry in secondary_care_repeat_dispensing_process_request['entry']:
+        resource = entry["resource"]
+        if resource["resourceType"] == "Provenance":
+            for signature in resource["signature"]:
+                signature_time_updated = signature["when"]
+                break
+    assert signature_time == signature_time_updated
+
+
+def test_update_prescription__updates_authored_on(secondary_care_repeat_dispensing_process_request):
+    authored_on = "newValue"
+    update_prescription(secondary_care_repeat_dispensing_process_request, None, None, authored_on, None)
+    for entry in secondary_care_repeat_dispensing_process_request['entry']:
+        resource = entry["resource"]
+        if resource["resourceType"] == "MedicationRequest":
+            authored_on_updated = resource["authoredOn"]
+    assert authored_on == authored_on_updated
+
+
+def test_update_prescription__sets_validity_period_for_4_weeks_from_today(
+    secondary_care_repeat_dispensing_process_request
+):
+    update_prescription(secondary_care_repeat_dispensing_process_request, None, None, None, None)
+    for entry in secondary_care_repeat_dispensing_process_request['entry']:
+        resource = entry["resource"]
+        if resource["resourceType"] == "MedicationRequest":
+            if "validityPeriod" in resource["dispenseRequest"]:
+                validityStart = resource["dispenseRequest"]["validityPeriod"]["start"]
+                validityEnd = resource["dispenseRequest"]["validityPeriod"]["end"]
+    assert validityStart == date.today().isoformat()
+    assert validityEnd == (date.today() + timedelta(weeks=4)).isoformat()
+
+
+def test_get_organisation_code__gets_org_code_from_secondary_care(secondary_care_repeat_dispensing_process_request):
+    organisationCode = get_organisation_code(secondary_care_repeat_dispensing_process_request)
+    for entry in secondary_care_repeat_dispensing_process_request['entry']:
```

Comment:
The code for the test duplicates the code being tested. Same for the primary care equivalent. If we know which message we're going to use for the test, can we just put the expected value in the test as a literal?

Reply:
We still have to traverse the python dict to read the values as we're checking that they were updated to new values so that code ends up being almost the same code as the update itself

Reply:
Ah yeah forgot our examples aren't static. Which is odd given the code I'm reviewing...